core/spec/models/spree/order_contents_spec.rb

```diff
@@ -224,6 +224,72 @@
       }.to change { subject.order.total }
     end
 
+    context "given an order with existing addresses" do
```

Comment:
These specs are great, and even if we decide to move the logic somewhere else, they should absolutely stay.